tests/migrations/test_operations.py

```diff
@@ -2653,89 +2653,116 @@ def create_data(models, schema_editor):
+                project_state = new_state
+                new_state = new_state.clone()
+                with connection.schema_editor() as editor:
+                    fill_initial_data.state_forwards("fill_initial_data", new_state)
+                    fill_initial_data.database_forwards("fill_initial_data", editor, project_state, new_state)
 
-        project_state = new_state
-        new_state = new_state.clone()
-        with connection.schema_editor() as editor:
-            grow_article_id.state_forwards("test_article", new_state)
-            grow_article_id.database_forwards("test_article", editor, project_state, new_state)
+                project_state = new_state
+                new_state = new_state.clone()
+                with connection.schema_editor() as editor:
+                    grow_article_id.state_forwards("test_article", new_state)
+                    grow_article_id.database_forwards("test_article", editor, project_state, new_state)
 
-        state = new_state.clone()
-        article = state.apps.get_model("test_article.Article")
-        self.assertIsInstance(article._meta.pk, models.BigAutoField)
+                state = new_state.clone()
+                article = state.apps.get_model("test_article.Article")
+                self.assertIsInstance(article._meta.pk, target_field)
 
-        project_state = new_state
-        new_state = new_state.clone()
-        with connection.schema_editor() as editor:
-            grow_blog_id.state_forwards("test_blog", new_state)
-            grow_blog_id.database_forwards("test_blog", editor, project_state, new_state)
+                project_state = new_state
+                new_state = new_state.clone()
+                with connection.schema_editor() as editor:
+                    grow_blog_id.state_forwards("test_blog", new_state)
+                    grow_blog_id.database_forwards("test_blog", editor, project_state, new_state)
 
-        state = new_state.clone()
-        blog = state.apps.get_model("test_blog.Blog")
-        self.assertIsInstance(blog._meta.pk, models.BigAutoField)
+                state = new_state.clone()
+                blog = state.apps.get_model("test_blog.Blog")
+                self.assertIsInstance(blog._meta.pk, target_field)
 
-        project_state = new_state
-        new_state = new_state.clone()
-        with connection.schema_editor() as editor:
-            fill_big_data.state_forwards("fill_big_data", new_state)
-            fill_big_data.database_forwards("fill_big_data", editor, project_state, new_state)
+                project_state = new_state
+                new_state = new_state.clone()
+                with connection.schema_editor() as editor:
+                    fill_big_data.state_forwards("fill_big_data", new_state)
+                    fill_big_data.database_forwards("fill_big_data", editor, project_state, new_state)
+
+                delete_article = migrations.DeleteModel("Article")
```

Comment:
Do we need to delete these models? I don't think that below tests are required,

Reply:
Yes, because of looping and using `self.subTest()`. Otherwise it complains that they already exist from the second pass through.

Reply:
I added `_test_autofield_foreignfield_growth()` to reduce diff and make this more readable. `subTest()` is not really handy with schema tests because it can cause isolation issues.